### PR TITLE
adding e2e tests for ServiceBus connection exhaustion

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

File: test/WebJobs.Script.Tests.E2E/GeneralEndToEndTests.cs [hunk 1]
```diff
@@ -136,14 +138,63 @@ public GeneralEndToEndTests(FunctionAppFixture fixture)
...
+            {
+                // max connections in dynamic is currently 300
+                for (i = 1; i <= 25; i++)
+                {
+                    for (j = 1; j <= 25; j++)
+                    {
+                        List<Task<HttpResponseMessage>> requestTasks = new List<Task<HttpResponseMessage>>();
+                        requestTasks.Add(client.GetAsync($"api/ServiceBusNode?code={_fixture.FunctionDefaultKey}"));
+
+                        await Task.WhenAll(requestTasks);
```
Comment (fabiocav) [inline test/WebJobs.Script.Tests.E2E/GeneralEndToEndTests.cs]: Did you want this `WhenAll` to be outside of the scope of this loop? We'll only have one task here at this point.

Comment (brettsam) [inline test/WebJobs.Script.Tests.E2E/GeneralEndToEndTests.cs]: Umm. Yeah. :-)

**Comment by brettsam** [comment]
Well, when the fix is, this should be rock-solid and it shouldn't matter. Since these tests won't block checkin, I'd opt for leaving it like this and if we start to see odd failures that may be related, we can dig in. It may actually bring something to light that we normally don't see.

**Comment by fabiocav** [comment]
Sounds good!

**Pull request merged**